databricks_cli/stack/api.py

```diff
@@ -0,0 +1,329 @@
+        the resource along with the physical id and deploy output of the resource.
+        """
+        try:
+            resource_id = resource[RESOURCE_ID]
+            resource_type = resource[RESOURCE_TYPE]
+            resource_properties = resource[RESOURCE_PROPERTIES]
+        except KeyError as e:
+            raise StackError("%s doesn't exist in resource config" % str(e))
+
+        # Deployment
+        physical_id = self._get_deployed_resource(resource_id, resource_type)
+
+        if resource_type == JOBS_TYPE:
+            physical_id, deploy_output = self.deploy_job(resource_id, resource_properties,
+                                                         physical_id)
+        else:
+            raise StackError("Resource type '%s' not found" % resource_type)
+
+        resource_deploy_info = {RESOURCE_ID: resource_id, RESOURCE_TYPE: resource_type}
+        if six.PY3:
+            resource_deploy_info[RESOURCE_DEPLOY_TIMESTAMP] = datetime.now().timestamp()
+        elif six.PY2:
+            resource_deploy_info[RESOURCE_DEPLOY_TIMESTAMP] = \
+                time.mktime(datetime.now().timetuple())
+        resource_deploy_info[RESOURCE_PHYSICAL_ID] = physical_id
+        resource_deploy_info[RESOURCE_DEPLOY_OUTPUT] = deploy_output
+        return resource_deploy_info
+
+    def deploy(self, filename):  # overwrite to be added
+        """
+        Deploys a stack given stack JSON configuration template at path filename.
+
+        Loads the JSON template as well as status JSON if stack has been deployed before.
+        After going through each of the resources and deploying them, stores status JSON
+        of deployment with deploy status of each resource deployment.
+
+        :param filename: Path to stack JSON configuration template
+        :return: None.
+        """
+        config_filepath = os.path.abspath(filename)
+        config_dir = os.path.dirname(config_filepath)
+        cli_cwd = os.getcwd()
+        os.chdir(config_dir)  # Switch current working directory to where json is stored
+        try:
+            parsed_conf = self._parse_config_file(filename)
+            stack_name = parsed_conf['name'] if 'name' in parsed_conf else None
+
+            self._load_deploy_metadata(config_filepath)
+
+            deploy_metadata = {STACK_NAME: stack_name, 'cli_version': CLI_VERSION}
+            click.echo('Deploying stack %s' % stack_name)
+            deploy_metadata[STACK_RESOURCES] = parsed_conf[STACK_RESOURCES]
+            deployed_resources = []
+            if STACK_RESOURCES not in parsed_conf:
+                raise StackError("'%s' not in configuration" % STACK_RESOURCES)
+            for resource in parsed_conf[STACK_RESOURCES]:
+                click.echo()
+                click.echo("Deploying resource")
+                deploy_status = self.deploy_resource(resource)  # overwrite to be added
+                if deploy_status:
```

Comment:
Please use consistent naming. It seems that `deploy_status` is the samething as `stack_deploy_info` previously. There are also other similar related objects such as `deploy_output`, `deployed_metadata` etc.

Reply:
How about using `stack_status` for the output file. and `resource_status` for the output of `deploy_resource` method?